src/puzzle/KeyInput.js

```diff
@@ -56,6 +56,14 @@ pzpr.classmgr.makeCommon({
 			var c = this.getchar(e);
 			if (!this.enableKey) {
 				if (c === "BS" || c === " ") {
+					var el = document.activeElement;
+					if (
+						(el.tagName === "INPUT" &&
+							(el.type === "number" || el.type === "text")) ||
+						el.tagName === "TEXTAREA"
+					) {
+						return;
+					}
```

Comment:
So let's see if I understand this correctly:

- `enableKey` should be `false` precisely when some pop-up is open (with the meaning that keyboard input on the puzzle itself is disabled while a popup is up)
- in that case, the puzzle itself plain ignores the key event, except for backspace and space, which it prevents from being handled elsewhere
- this change makes sure that this ignoring doesn't happen on certain elements

Some thoughts:
- if we stick with this, the same change should happen in `e_keyup`
- it would be clearer what's going on if the common check was a method with a clear name
- This all doesn't really seem to belong inside the puzzle applet itself, but rather should be handled within the ui (`src-ui`). I.e., the `"BS"` check shouldn't really be here at all. Do you see a way to lift it to the ui?

Reply:
(I opened a thread in the zulip)